### PR TITLE
Replace angular2 template with latest version of angular

### LINKED ISSUE #43: Update angular template to v4

### PR DESCRIPTION
**Pull request by ebordon**
* [x ] I have read the [contribution documentation](https://github.com/electron-userland/electron-forge-templates/blob/master/CONTRIBUTING.md) for this project.
* [x ] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] The changes are appropriately documented (if applicable).
* [x] The changes have sufficient test coverage (if applicable).
* [x ] The testsuite passes successfully on my local machine (if applicable).

**Summarize your changes:**

Updated angular2 templates to the up to date angular packages.

```
"dependencies": {
  "@angular/common": "^4.3.4",
  "@angular/compiler": "^4.3.4",
  "@angular/core": "^4.3.4",
  "@angular/platform-browser": "^4.3.4",
  "@angular/platform-browser-dynamic": "^4.3.4",
  "electron-compile": "^6.4.1",
  "electron-devtools-installer": "^2.2.0",
  "reflect-metadata": "^0.1.10",
  "tslib": "^1.7.1",
  "zone.js": "^0.8.16"
},
"devDependencies": {
  "babel-plugin-transform-async-to-generator": "^6.24.1",
  "babel-preset-env": "^1.6.0",
  "babel-preset-react": "^6.24.1",
  "electron-prebuilt-compile": "1.7.5",
  "tslint": "^4.5.1",
  "typescript": "^2.4.2"
}
```

Fixes #43.

### CODE REVIEW

**Review by malept — DISMISSED** [review]
Thanks for the PR! Could you go ahead and rename the NPM package and folder to angular4 as well?

**Comment by ebordon** [comment]
Done the requested changes. Version of the package still doing the same.

**Review by ebordon — COMMENTED** [review]
Done the requested changes!

**Comment by malept** [comment]
@ebordon does this fix electron-userland/electron-forge#272?

**Comment by malept** [comment]
Also, electron-userland/electron-forge#301, which the reporter asserts is different from electron-userland/electron-forge#272. (I don't know, because I'm not familiar with Angular.)

**Comment by ebordon** [comment]
@malept both issues you mentioned are templates reference errors. They have to correct the path in the component declaration using src/ folder as root.

I've commented on both issues.

**Comment by malept** [comment]
Thanks. At this point I'd like @MarshallOfSound to take a quick look before we merge.

**Comment by ebordon** [comment]
Is there no news about this?

Cheers

**Comment by malept** [comment]
I'd like @MarshallOfSound to take a look at it mostly because I'd like to discuss what we want to do about the existing angular2 template and the future angular4 template. Do we deprecate it in NPM? Do we start over versioning at 1.0.0? Things like that.

**Comment by schie** [comment]
Hello everyone, is there any more movement on this?

**Comment by alexeagle** [comment]
Angular 5 is in RC and will be released in the next few weeks. Will we just update the templates again, or is it something the user should be able to control?

Even better, you have `react`, not `react16`. Angular 1 to 2 was a rewrite, but future updates will not be. So it should just be `angular`. Otherwise you'll have this same issue every 6 months.

**Comment by malept** [comment]
Should we just rename this to `angular` instead of `angularX`? I haven't been paying attention to how Angular versions things these days.

**Comment by MarshallOfSound** [comment]
Sorry guys, this slipped off my radar.  I believe we should just rename `angular2` to `angular` as afaik any major bumps to angular from now on aren't super massive changes like 1->2 was.  We should be safe treating it the same as we treat react 👍

**Comment by alexeagle** [comment]
I looked into this a bit yesterday.

There is a small amount of electron templating (a section in the
package.json and the .compilerc file) and a large amount of angular
templating (a sample app, package.json with peer dependency versions too).

I think maybe a better approach is to run @angular/cli as the scaffold
generator rather than the electron-forge-templates one, then just add the
missing file and patch up the package.json file. This also means you
generate an app that is compatible with the Angular CLI; today the
generated angular-electron project is missing this key feature.

**Comment by malept** [comment]
My concern about integrating Angular CLI is that it would make `@angular/cli` an indirect dependency of Forge, regardless of if you use Angular.

**Comment by alexeagle** [comment]
Yeah I wanted to create a mostly-empty project in your template with
@angular/cli in the devDependencies of that app, then call
./node_modules/.bin/ng new --directory=.
Except there is no such option for the CLI to create the app in an existing
directory. Maybe I can add that option.

**Comment by malept** [comment]
Just so I'm clear, we're not opposed to integrating with the Angular CLI (that's why we've kept https://github.com/electron-userland/electron-forge/issues/262 open - we want to work with the Angular community to get this right), we just need to figure out a way to do it in a way that doesn't negatively impact other communities that use Forge.

**Comment by alexeagle** [comment]
I totally agree that @angular/cli shouldn't be a dependency for non-angular
users, I think we should consider changes to angular CLI to support
electron rather than assuming we must scaffold out based on existing release

**Comment by malept** [comment]
Ideally, it would be great if our support for Angular CLI was similar to the Ember CLI (in that it's just [a plugin](https://github.com/felixrieseberg/ember-electron) that utilizes Forge internally), but I'm not familiar with the Angular CLI architecture so I don't know how feasible it is.

**Comment by chadbr** [comment]
Would this help at all?

https://github.com/sean-perkins/angular-electron-seed

**Comment by malept** [comment]
I've rebased the PR on master, updated the dependencies to Angular 5, and renamed the package to `electron-forge-templates-angular`. However, I don't use Angular and I don't currently have a stable internet connection, so I'd appreciate it if someone could test this. Instructions (`hub` refers to Github's [`hub`](https://github.com/github/hub)):

1. `hub clone electron-userland/electron-forge-templates && cd electron-forge-templates`
2. `hub checkout https://github.com/electron-userland/electron-forge-templates/pull/47`
3. `electron-forge init --template=angular /path/to/new/electron/app`

**Comment by alexeagle** [comment]
Angular 6 is due for release in a week or two, maybe we should just target
that?

**Comment by malept** [comment]
We can do that too.

**Comment by alexeagle** [comment]
There is a new affordance in Angular CLI called `ng add` which is intended for exactly this sort of thing - take some third-party library/framework and add its configuration files to my project, using this https://www.npmjs.com/package/@angular-devkit/schematics
I know it's a bigger scope, but if we want to make electron a good target for average Angular users, that's the most ergonomic way.

File: angular/index.js [hunk 1]
```diff
@@ -3,20 +3,21 @@ const path = require('path');
...
-    '@angular/platform-browser-dynamic@^2.4.1',
-    'electron-devtools-installer@^2.1.0',
-    'reflect-metadata@^0.1.9',
-    'tslib@^1.4.0',
-    'zone.js@^0.7.4'
+    '@angular/common@^5.2.9',
+    '@angular/compiler@^5.2.9',
+    '@angular/core@^5.2.9',
+    '@angular/platform-browser@^5.2.9',
+    '@angular/platform-browser-dynamic@^5.2.9',
```
Comment (malept) [inline angular/index.js:10]: Is it even worth pinning Angular to a major version at this point?

**Comment by gate3** [comment]
I made a scaffold for angular5. How do i contribute? You can find it here https://github.com/gate3/Electron-forge-angular5

**Comment by chriszrc** [comment]
Any updates on this? As of version 6 of the angular cli, it now provides a way to automatically update your projects, which maybe could mean that as long as electron-forge starts off with v6, users could update to whatever the latest version is?

https://angular.io/cli/update

**Comment by malept** [comment]
Closing, see https://github.com/electron-userland/electron-forge-templates/issues/66

Please note that the Forge developers are still interested in integration with framework CLIs like we have with `ember-electron`. If you would like to help us with that, please reach out to us in the Electron Forge issue tracker.

**Pull request closed without merging**